File: frontend/src/app/features/epargne/models/compte-epargne-create-request.ts
import { TypeCompteEpargne } from "./type-compte-epargne";

export type StatutCompte = 'ACTIF' | 'INACTIF' | 'FERME' | 'BLOQUE';

export interface CompteEpargneCreateRequest {
  membreId: number;
  typeCompte: TypeCompteEpargne;
  dateOuverture: string; // yyyy-MM-dd
}

File: backend/src/main/java/com/mini/credit/dto/epargne/CompteEpargneCreateRequest.java
package com.mini.credit.dto.epargne;

import com.mini.credit.enums.TypeCompteEpargne;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
public class CompteEpargneCreateRequest {

    @NotNull
    private Long membreId;

    @NotNull
    private TypeCompteEpargne typeCompte;

    @NotNull
    private LocalDate dateOuverture;
}
